### ISSUE TITLE
Link component preserve-scroll attribute not working for composition API

### ISSUE THREAD

**Issue by matthewknill** [inertiajs/inertia#1461]
### Version:

- `@inertiajs/vue3` version: 1.0.2

### Describe the problem:

When using the Inertia `router` with the `{ preserveScroll: true }` option, the scroll is preserved as expected. However, when using the `preserve-scroll` attribute on the `<Link>` component, it scrolls to the top... I have noticed this when I migrated from the _options API_ to the _composition API_. I reverted back to using the _options API_ for one of my components and it seems to work as expected.

The following is what I'm using in my pagination component which when clicked scrolls to the top of the page:
```vue
<Link v-else :key="`link-${key}`" class="mb-1 mr-1 px-4 py-3 focus:text-indigo-500 text-sm leading-4 hover:bg-white border rounded" :class="{ 'bg-white': link.active }" :href="link.url" preserve-scroll v-html="link.label" />
```

### Steps to reproduce:

Clicking on the link component...

**Comment by xtreme2020** [comment]
Hi,
Try importing from:
 import { Link } from '@inertiajs/inertia-vue3';   

//For composition API i'm importing everything from @inertiajs/inertia-vue3 to get it working

Don't know why but there are some issues combining compositions API and latest router version

**Comment by matthewknill** [comment]
Hmm when I import using `import { Link } from '@inertiajs/inertia-vue3'` it hard loads the page (i.e. I can see it reload the page on the browser).

**Comment by matthewknill** [comment]
Closed in place of: #1538
